sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTestCommon.java

```diff
@@ -391,4 +391,35 @@ void testWriteWithTypeFn() throws Exception {
       assertEquals(type + " holds incorrect count", adjustedNumDocs / 2, count);
     }
   }
+
+  /**
+   * Tests that documents are correctly routed when index, type and document ID functions are
+   * provided to overwrite the defaults of using the configuration and auto-generation of the
+   * document IDs by Elasticsearch. The scientist name is used for the index, type and document ID.
+   * As a result there there should be only a single document in each index/type.
```

Comment:
-there

Reply:
what is the overhead running time of this test comparing to testWriteInVolume on your local machine in percentage?

Reply:
2.6% increase.  Raw data below:

```
<testcase name="testWriteVolume" classname="org.apache.beam.sdk.io.elasticsearch.ElasticsearchIOIT" time="10.537"/>
<testcase name="testWriteWithFullAddressingVolume" classname="org.apache.beam.sdk.io.elasticsearch.ElasticsearchIOIT" time="10.816"/>
```